gradio/helpers.py

```diff
@@ -274,7 +274,16 @@ async def load_example(example_id):
                 )
 
         if self.cache_examples:
-            await self.cache()
+            if wasm_utils.IS_WASM:
+                # In the Wasm mode, the `threading` module is not supported,
+                # so `client_utils.synchronize_async` is also not available.
+                # And `self.cache()` should be waited for to complete before this method returns,
+                # (otherwise, an error "Cannot cache examples if not in a Blocks context" will be raised anyway)
+                # so `eventloop.create_task(self.cache())` is also not an option.
+                raise wasm_utils.WasmUnsupportedError(
+                    "Caching examples is not supported in the Wasm mode."
+                )
+            client_utils.synchronize_async(self.cache)
```

Comment:
This updated code is more consistent to `chat_interface.py` below rather than `await` 👍
https://github.com/gradio-app/gradio/blob/1265a9ac134f867f4187c1f7d15ab27dbdf7503b/gradio/chat_interface.py#L237-L238

Reply:
Yep this should be fine. Let me just test to confirm